cinema/views.py

```diff
@@ -43,16 +66,111 @@
 
         return MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset.prefetch_related()
+
+        actors = self.request.query_params.get("actors")
+        if actors:
+            actor_values = actors.split(",")
+            query = Q()
+            for value in actor_values:
+                if value.isdigit():
+                    query |= Q(id=value)
+                else:
+                    query |= Q(actors__first_name__icontains=value) | Q(
+                        actors__last_name__icontains=value
+                    )
+            queryset = queryset.filter(query)
+
+        title = self.request.query_params.get("title")
+        if title:
+            queryset = queryset.filter(
+                title__icontains=title
+            )
+
+        genres = self.request.query_params.get("genres")
+        if genres:
+            genres_list = genres.split(",")
+            print("Genres list:", genres_list)
+            queryset = queryset.filter(
+                genres__id__in=genres_list
+            )
+
+        return queryset.distinct()
+
+
+from django.db.models import F, Count
+from django.db.models.functions import TruncDate
```

Comment:
The imports for `F`, `Count`, and `TruncDate` are duplicated. Remove these redundant imports to clean up the code.